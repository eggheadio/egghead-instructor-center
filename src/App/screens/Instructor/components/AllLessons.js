import {Component, PropTypes} from 'react'
import {connect} from 'react-redux'
import {requestAllLessons} from '../state/actions'

export default connect(
  null,
  {requestAllLessons}
)(class AllLessons extends Component {
  
  static propTypes = {
    children: PropTypes.func.isRequired,
  }

  static defaultProps = {
    size: 20,
    states: undefined,
  }

  state = {
    currentPage: 1
  }

  fetchLessons = (currentPage = 1) => {
    const {
      requestAllLessons,
      size,
      states,
    } = this.props
    this.setState({currentPage})
    requestAllLessons({
      page: currentPage,
      size,
      states,
    })
  }

  componentWillMount() {
    this.fetchLessons()
  }

  render() {
    const {currentPage} = this.state
    const {children, size} = this.props
    return children(currentPage, this.fetchLessons, size)
  }
})
