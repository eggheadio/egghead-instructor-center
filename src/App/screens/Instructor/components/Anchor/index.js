import React, {PropTypes} from 'react'

const Anchor = ({
  url,
  children,
}) => (
  <a
    href={url}
    target='_blank'
    className='blue'
  >
    {children}
  </a>
)

Anchor.propTypes = {
  url: PropTypes.string.isRequired,
  children: PropTypes.node,
}

export default Anchor
