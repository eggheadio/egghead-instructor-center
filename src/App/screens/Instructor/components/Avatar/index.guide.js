import React from 'react'
import {storiesOf} from '@kadira/storybook'
import faker from 'faker'
import Avatar from '.'

storiesOf('Avatar', module)
  .add('default', () => (
    <Avatar
      name={faker.name.firstName()}
      url={faker.image.avatar()}
    />
  ))
