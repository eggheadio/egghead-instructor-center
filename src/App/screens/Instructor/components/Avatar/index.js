import React from 'react'

const Avatar = ({name, url}) => (
  <img
    alt={`Avatar for ${name}`}
    src={url}
    className='br-100'
  />
)

Avatar.propTypes = {
  name: React.PropTypes.string.isRequired,
  url: React.PropTypes.string.isRequired,
}

export default Avatar
