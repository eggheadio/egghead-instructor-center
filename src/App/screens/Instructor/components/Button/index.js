import React from 'react'

export default ({children, onClick, className}) => (
  <button
    className={`button-reset bn dib b0 ttu dim br2 ph3 pv2 white bg-orange tc ${className}`}
    onClick={onClick}
  >
    {children}
  </button>
)
