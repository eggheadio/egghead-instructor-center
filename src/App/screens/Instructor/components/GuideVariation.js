import React, {PropTypes} from 'react'

const GuideVariation = ({tooltip, children}) => (
  <div title={tooltip}>
    {children}
  </div>
)

GuideVariation.propTypes = {
  tooltip: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default GuideVariation

