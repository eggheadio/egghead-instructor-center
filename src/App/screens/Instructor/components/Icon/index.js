import React, {PropTypes} from 'react'
import {keys} from 'lodash'

export const sizes = {
  1: '',
  2: 'fa-lg',
  3: 'fa-2x',
  4: 'fa-3x',
  5: 'fa-4x',
  6: 'fa-5x',
}

export const types = {
  'menu': 'bars',
  'step-incomplete': 'square-o',
  'step-complete': 'check-square-o',
  'close': 'close',
  'more-info': 'info-circle',
  'subscriber-minutes': 'clock-o',
  'revenue': 'money',
  'course': 'folder-open-o',
  'lesson': 'file-o',
}

const Icon = ({
  type,
  size = '1',
  className = '',
}) => (
  <span className={`
    fa
    fa-${types[type]} 
    ${sizes[size]}
    ${className}
  `} />
)

Icon.propTypes = {
  type: PropTypes.oneOf(keys(types)).isRequired,
  size: PropTypes.oneOf(keys(sizes)),
  className:  PropTypes.string,
}

export default Icon
