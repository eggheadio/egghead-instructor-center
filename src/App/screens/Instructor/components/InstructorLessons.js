import {Component, PropTypes} from 'react'
import {connect} from 'react-redux'
import {requestInstructorLessons} from '../state/actions'

export default connect(
  null,
  {requestInstructorLessons}
)(class InstructorLessons extends Component {
  
  static propTypes = {
    children: PropTypes.func.isRequired,
  }

  state = {
    currentPage: 1
  }

  fetchLessons = (currentPage = 1) => {
    const {
      requestInstructorLessons,
      instructor,
      size = 10,
      states = undefined,
    } = this.props
    this.setState({currentPage})
    requestInstructorLessons({
      lessons_url: instructor.lessons_url,
      page: currentPage,
      size,
      states,
    })
  }

  componentWillMount() {
    this.fetchLessons()
  }

  render() {
    const {currentPage} = this.state
    const {children} = this.props
    return children(currentPage, this.fetchLessons)
  }
})
