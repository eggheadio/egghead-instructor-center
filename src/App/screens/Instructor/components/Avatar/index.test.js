import React from 'react'
import renderer from 'react-test-renderer'
import {StringFixture, ImageUrlFixture} from '../../../../components/SnapshotFixtures'
import Avatar from '.'

test('default', () => {
  const tree = renderer.create(
    <Avatar
      name={StringFixture}
      url={ImageUrlFixture}
    />
  )
  expect(tree).toMatchSnapshot()
})
