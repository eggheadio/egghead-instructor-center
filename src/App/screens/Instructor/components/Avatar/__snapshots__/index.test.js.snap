exports[`test default 1`] = `
<img
  alt="Avatar for String"
  className="br-100"
  src="https://s3.amazonaws.com/uifaces/faces/twitter/kimcool/128.jpg" />
`;
